template2/build.sh

```diff
@@ -0,0 +1,97 @@
+    echo "       $0 [-r] [-u <URL>] [--] [arguments to msbuild]"
+    echo "       $0 [-r] [-b <BRANCH>] [--] [arguments to msbuild]"
+    echo ""
+    echo "Arguments:"
+    echo "      -r, --reset-korebuild               Delete the current `.build` directory and re-fetch KoreBuild"
+    echo "      -u, --korebuild-url <URL>           Fetch KoreBuild from URL"
+    echo "      -b, --korebuild-branch <BRANCH>     Fetch KoreBuild from BRANCH in the default repository (https://github.com/aspnet/KoreBuild)"
+    echo "      --                                  Consider all remaining arguments arguments to MSBuild when building the repo."
+    echo ""
+    echo "Notes:"
+    echo "      The '--' switch is only necessary when you want to pass an argument that would otherwise be recognized by this"
+    echo "      script to MSBuild. By default, any unrecognized argument will be forwarded to MSBuild."
+    echo ""
+    echo "      If you wish to build a specific target from the MSBuild project file, use the '-t:<TARGET>' switch, which will be forwarded"
+    echo "      to MSBuild. For example `.\build.sh -t:Verify`"
+}
+
+while [[ $# > 0 ]]; do
+    case $1 in
+        -h|-\?|--help)
+            show_help
+            exit 0
+            ;;
+        -r|--reset-korebuild)
+            KOREBUILD_RESET=1
+            ;;
+        -u|--korebuild-url)
+            KOREBUILD_URL=$2
+            shift
+            ;;
+        -b|--korebuild-branch)
+            KOREBUILD_BRANCH=$2
+            shift
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *)
+            break
+            ;;
+    esac
+    shift
+done
+
+if [ -z $KOREBUILD_URL ]; then
+    if [ ! -z $KOREBUILD_BRANCH ]; then
+        KOREBUILD_URL="https://github.com/aspnet/KoreBuild/tarball/$KOREBUILD_BRANCH"
+    else
+        KOREBUILD_URL="https://github.com/aspnet/KoreBuild/tarball/anurse/msbuild"
+    fi
+fi
+
+BUILD_FOLDER="$DIR/.build"
+BUILD_FILE="$BUILD_FOLDER/build2/KoreBuild.sh"
+
+if [[ -d $BUILD_FOLDER && $KOREBUILD_RESET = "1" ]]; then
+    echo "Cleaning old KoreBuild folder to force a reset ..."
+    rm -Rf $BUILD_FOLDER
+fi
+
+if [ ! -d $BUILD_FOLDER ]; then
+    echo "Downloading KoreBuild from $KOREBUILD_URL ..."
+
+    KOREBUILD_DIR=`mktemp -d`
+    KOREBUILD_TAR="$KOREBUILD_DIR/korebuild.tar.gz"
+
+    retries=6
+    until (wget -O $KOREBUILD_TAR $KOREBUILD_URL 2>/dev/null || curl -o $KOREBUILD_TAR --location $KOREBUILD_URL 2>/dev/null); do
+        echo "Failed to download '$KOREBUILD_TAR'"
+        if [ "$retries" -le 0 ]; then
+            exit 1
+        fi
+        retries=$((retries - 1))
+        echo "Waiting 10 seconds before retrying. Retries left: $retries"
+        sleep 10s
+    done
+
+    # Sub-shell (to preserve current dir)
+    mkdir $BUILD_FOLDER
+    cd $BUILD_FOLDER
+    tar xf $KOREBUILD_TAR --strip-components 1
+    rm -Rf $KOREBUILD_DIR
+fi
+
+cd $DIR
+chmod a+x $BUILD_FILE
+$BUILD_FILE "$@"
```

Comment:
No, I tested it out and it will pass each argument from the current `$1` on, so the `shift` calls above remove the leading KoreBuild.sh arguments from that list.